src/ssl.ml

```diff
@@ -42,6 +42,9 @@ type ssl_error =
   | Error_zero_return
   | Error_want_connect
   | Error_want_accept
+  | Error_want_async
+  | Error_want_async_job
+  | Error_want_client_hello_cb
```

Comment:
should we also add [SSL_ERROR_WANT_RETRY_VERIFY](https://github.com/openssl/openssl/blob/0bf7e94c10f1b00510b8a36cdcbedc02a66468be/include/openssl/ssl.h.in#L1217) for completeness?

Reply:
I do not have it in my openssl.h. It is the next one on the list (the order matters here ?)

Reply:
This list must be complete to the latest version of openssl, a missing case could mean a segfault ?

Reply:
It is added, does not raise any problem when you use an older version of openssl. I also added a safer conversion [9f00f5a](https://github.com/savonet/ocaml-ssl/pull/117/commits/9f00f5ab0b222b9a3f98b227d6ea4c4c3d018b49) for `Ssl.ssl_error` to make sure we have no problem if we do not follow the evolution of open_ssl.

Reply:
I'm not sure how much safer that actually is, since now one must remember to update this value 13 when we add a variant case next.

Reply:
My test avoid all possible missing case in pattern matching. If we forget to increade 13, it will still be true, we will just not have the wanted error. May be a comment in the type ? Otherwise we let as it was ?

Reply:
yeah perhaps revert.

Reply:
I added a comment, but it should be in another PR and should be for other type too.
I just reverted.

 The correct direction is 
- either to generate the ocaml data type
- either generate some .h from the ocaml data type.

What do you prefer ?

Reply:
Still think missing cases in ocaml data type is a big problem.
Don't you think so ?

Reply:
I'm a bit uncertain what to do, maybe open an issue where we can discuss.

The problem with capping the number of variants is that (in our specific case) 13 could mean "unknown" or "new enum that openssl added" in the future.

Reply:
> Still think missing cases in ocaml data type is a big problem. Don't you think so ?

Not necessarily:

- Most errors are covered already
- they represent exceptional behavior in most, if not all, programs
- the OpenSSL model is not likely to change significantly in a way that adds a few more of these